rust/lance/src/dataset/scanner.rs

```diff
@@ -522,15 +573,101 @@ impl Scanner {
             extra_columns.push(vector_field);
             extra_columns.push(ArrowField::new(DIST_COL, DataType::Float32, true));
         };
+
         if self.with_row_id {
             extra_columns.push(ROW_ID_FIELD.clone());
         }
 
         let schema = if !extra_columns.is_empty() {
-            self.projections.merge(&ArrowSchema::new(extra_columns))?
+            self.phyical_columns
+                .merge(&ArrowSchema::new(extra_columns.clone()))?
         } else {
-            self.projections.clone()
+            self.phyical_columns.clone()
         };
+
+        // drop metadata
+        // NOTE: this is the current behavior as we don't return metadata in queries
+        // We should reconsider this behavior
```

Comment:
I'm not sure I agree that we should preserve it.  Metadata is a table/dataset property.  It doesn't make sense on individual batches.

Reply:
I agree with that. Let me reword the comment. What I meant is that queries today don't return metadata, but scans do. We should make this behavior consistent